src/main/java/com/study/realworld/security/JjwtService.java

```diff
@@ -0,0 +1,73 @@
+package com.study.realworld.security;
+
+import com.study.realworld.user.domain.User;
+import com.study.realworld.user.domain.UserRepository;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.MalformedJwtException;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.UnsupportedJwtException;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import java.security.Key;
+import java.util.Date;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JjwtService implements JwtService {
+
+    private final Key key;
+
+    private final String headerType;
+
+    private final String issuer;
+
+    private final long accessTime;
+
+    private final UserRepository userRepository;
+
+    public JjwtService(@Value("${jwt.token.header-type}") String headerType,
+        @Value("${jwt.token.issuer}") String issuer,
+        @Value("${jwt.token.secret}") String secret,
+        @Value("${jwt.token.access-time}") long accessTime,
+        UserRepository userRepository) {
+        byte[] keyBytes = Decoders.BASE64.decode(secret);
+        this.key = Keys.hmacShaKeyFor(keyBytes);
+        this.headerType = headerType;
+        this.issuer = issuer;
+        this.accessTime = accessTime;
+        this.userRepository = userRepository;
+    }
+
+    public String createToken(User user) {
+        return Jwts.builder()
+            .signWith(key, SignatureAlgorithm.HS512)
+            .setHeaderParam("typ", headerType)
```

Comment:
`typ` 매직 스트링보단 상수로 빼는게 좋을 것 같습니다
```suggestion
            .setHeaderParam(JWT_HEADER_PARAM_TYPE, headerType)
```